Source/Particles/Collision/BinaryCollision/Bremsstrahlung/BremsstrahlungFunc.H

```diff
@@ -0,0 +1,529 @@
+                w0 = (m_KEgrid_eV[index+1] - KErel_eV)/(m_KEgrid_eV[index+1] - m_KEgrid_eV[index]);
+            }
+
+            // kdsigdk at the cutoff
+            amrex::ParticleReal const w00 = (m_koT1_grid[i0_cut+1] - koT1_cut)/(m_koT1_grid[i0_cut+1] - m_koT1_grid[i0_cut]);
+            amrex::ParticleReal const w01 = 1.0_prt - w00;
+            kdsigdk_cut = (  w00*(w0*m_kdsigdk[index][i0_cut  ] + (1.0_prt - w0)*m_kdsigdk[index+1][i0_cut])
+                           + w01*(w0*m_kdsigdk[index][i0_cut+1] + (1.0_prt - w0)*m_kdsigdk[index+1][i0_cut+1]));
+
+            amrex::ParticleReal sigma_total;
+            if (KEcut_eV/KErel_eV <= m_koT1_cut_default) {
+                // Use precalculated value
+                sigma_total = w0*m_sigma_total[index] + (1.0_prt - w0)*m_sigma_total[index+1];
+            } else {
+                // Integrate the distribution using trapezoidal rule
+                amrex::ParticleReal koT1_grid_im1 = koT1_cut;
+                amrex::ParticleReal kdsigdk_im1 = kdsigdk_cut;
+                amrex::ParticleReal sigma = 0.0_prt;
+                for (int i=i0_cut+1; i < nkoT1; i++) {
+                    amrex::ParticleReal const kdsigdk_i = w0*m_kdsigdk[index][i] + (1.0_prt - w0)*m_kdsigdk[index+1][i];
+                    amrex::ParticleReal const dk = (m_koT1_grid[i] - koT1_grid_im1);
+                    amrex::ParticleReal const k_ave = (m_koT1_grid[i] + koT1_grid_im1)*0.5_prt;
+                    amrex::ParticleReal const dsigdk = (kdsigdk_i + kdsigdk_im1)*0.5_prt/k_ave;
+                    sigma = sigma + dsigdk*dk;
+                    koT1_grid_im1 = m_koT1_grid[i];
+                    kdsigdk_im1 = kdsigdk_i;
+                }
+                sigma_total = sigma;
+            }
+
+            return sigma_total;
+        }
+
+        /* \brief Generate the photon energy from the differential cross section
+         *
+         * \param[in] index the index for the electron energy grid
+         * \param[in] i0_cut the index below the cut off for the photon energy grid
+         * \param[in] koT1_cut the k of the photon energy cut off
+         * \param[in] kdsigdk_cut the ksigdk at the cut off
+         * \param[in] w0 the grid weighting at the cut off
+         * \param[in] sigma_total the total cross section
+         * \param[in] random_number the uniformly spaced random number
+         * \result the photon energy
+         */
+        AMREX_GPU_HOST_DEVICE AMREX_INLINE
+        amrex::ParticleReal
+        Photon_energy(int index, int i0_cut, amrex::ParticleReal koT1_cut, amrex::ParticleReal kdsigdk_cut,
+                      amrex::ParticleReal w0, amrex::ParticleReal sigma_total, amrex::ParticleReal random_number) const
+        {
+            using namespace amrex::literals;
+            amrex::ParticleReal koT1_grid_im1 = koT1_cut;
+            amrex::ParticleReal kdsigdk_im1 = kdsigdk_cut;
+            amrex::ParticleReal sigma = 0._prt;
```

Comment:
Works, too (ints or floats, cast is constexpr):
```suggestion
            amrex::ParticleReal sigma = 0_prt;
```